### PR TITLE
Add new options for group-based authorization, ACR and HTTP proxy

### PR DESCRIPTION
**Pull request by alindeman**
See the README changes for details on the options being added.

Feedback on the names of the config options, as well as the implementation, welcomed.

### CODE REVIEW

**Review by lstoll — APPROVED** [review]
option naming seems fine, lgtm

File: authenticator.go [hunk 1]
```diff
@@ -22,16 +26,52 @@ type authenticator struct {
...
+	// `groups` is used by default if not set.
+	GroupsClaimKey string
+
+	// AuthorizedGroups is a list of groups required for authentication to pass.
+	// A user must be a member of at least one of the groups in the list, if
+	// specified.
+	//
+	// If the list is empty, group membership is not required for authentication
+	// to pass.
+	AuthorizedGroups []string
```
Comment (karlhungus) [inline authenticator.go:41]: Makes sense for us, I wonder if we should consider `CheckClaimKey` `CheckClaimValues` for wider appeal, or maybe json path key value pairs, or if the complexity around requirements would be just too much

**Review by karlhungus — APPROVED** [review]

File: authenticator.go (hunk 1 above)
Comment (alindeman) [reply authenticator.go:41]: I think I'd like to defer it until we have more use cases, but I think you're right that if we see more use cases that require more one-off options like this, we should consider something bigger picture like JSON Path or even Rego.

**Pull request merged**